### ISSUE TITLE
2 x hasMany still needed for many-to-many self reference

### ISSUE THREAD

**Issue by eventhough** [sequelize/sequelize#3084]
I am creating a many-to-many self reference association. Sequelize is telling me that using hasMany has been deprecated: `Using 2 x hasMany to represent N:M relations has been deprecated. Please use belongsToMany instead`

However, I've tried using belongsToMany instead of hasMany and it doesn't create the correct relationships needed in the through table. Take this code for example (postgres, 2.0.0-rc8):

```javascript
var Sequelize = require('sequelize');
var sequelize = new Sequelize('postgres://root@localhost/database_bug');

var Person = sequelize.define('Person', {
  id: {
    type: Sequelize.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: Sequelize.STRING,
    allowNull: false
  }
});

Person.hasMany(Person, { as: 'Children', through: 'PersonChildren' });

Person.sequelize.sync({force:true}).then(function() {
  Person.build({ name: 'John Doe' }).save().then(function(father) {
    Person.build({ name: 'Jane Doe' }).save().then(function(daughter) {
      father.addChildren([daughter]).then(function() {
        father.getChildren().then(function(result) {
          console.log(result[0].name);
        });
      });
    });
  });
});
```

This creates the through table correctly with a unique composite key for `PersonId` and the `ChildId`:

```postgres
          Table "public.PersonChildren"
  Column   |           Type           | Modifiers 
-----------+--------------------------+-----------
 createdAt | timestamp with time zone | not null
 updatedAt | timestamp with time zone | not null
 PersonId  | integer                  | not null
 ChildId   | integer                  | not null
Indexes:
    "PersonChildren_pkey" PRIMARY KEY, btree ("PersonId", "ChildId")
Foreign-key constraints:
    "PersonChildren_ChildId_fkey" FOREIGN KEY ("ChildId") REFERENCES "People"(id) ON UPDATE CASCADE ON DELETE CASCADE
    "PersonChildren_PersonId_fkey" FOREIGN KEY ("PersonId") REFERENCES "People"(id) ON UPDATE CASCADE ON DELETE CASCADE
```

But if we make the one small change and switch `hasMany` to `belongsToMany` we get this schema which is totally different:

```postgres
          Table "public.PersonChildren"
  Column   |           Type           | Modifiers 
-----------+--------------------------+-----------
 createdAt | timestamp with time zone | not null
 updatedAt | timestamp with time zone | not null
 PersonId  | integer                  | not null
Indexes:
    "PersonChildren_pkey" PRIMARY KEY, btree ("PersonId")
Foreign-key constraints:
    "PersonChildren_PersonId_fkey" FOREIGN KEY ("PersonId") REFERENCES "People"(id)
```

Is this expected behavior? Thanks!

**Comment by mickhansen** [comment]
That does appear to be a bug.

**Comment by grabbou** [comment]
Had the same bug and switched over hasMany x2 recently.

**Comment by mickhansen** [comment]
@grabbou you had a self referential N:M aswell?

**Comment by grabbou** [comment]
@mickhansen well, didn't read through the initial issue to be fair. I think I had a different case (just typical N:M relation). Let me check that once again (over the weekend I am getting read of fixtures loader that does not support "belongsToMany" so I will be able to check it). Let's assume there's no issue as long as I am not going to create a new ticket.

**Comment by mickhansen** [comment]
Fix published in `sequelize@2.0.1`

**Comment by lolero** [comment]
Is there a way to make this association two-way so that one could execute

`daughter.getParents()`

and get a list of the parents of Jane Doe?
